itest/assets_test.go

```diff
@@ -838,6 +839,88 @@ func createAssetInvoice(t *testing.T, dstRfqPeer, dst *HarnessNode,
 	return resp.InvoiceResult
 }
 
+// assertInvoiceHtlcAssets makes sure the invoice with the given hash shows the
+// individual HTLCs that arrived for it and that they show the correct asset
+// amounts when decoded.
+func assertInvoiceHtlcAssets(t *testing.T, node *HarnessNode,
+	addedInvoice *lnrpc.AddInvoiceResponse, assetAmount uint64) {
+
+	ctxb := context.Background()
+	ctxt, cancel := context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+
+	invoice, err := node.InvoicesClient.LookupInvoiceV2(
+		ctxt, &invoicesrpc.LookupInvoiceMsg{
+			InvoiceRef: &invoicesrpc.LookupInvoiceMsg_PaymentAddr{
+				PaymentAddr: addedInvoice.PaymentAddr,
+			},
+		},
+	)
+	require.NoError(t, err)
+	require.NotEmpty(t, invoice.Htlcs)
+
+	t.Logf("Asset invoice: %v", toProtoJSON(t, invoice))
+
+	var totalAssetAmount uint64
+	for _, htlc := range invoice.Htlcs {
+		require.NotEmpty(t, htlc.CustomChannelData)
+
+		jsonHtlc := &rfqmsg.JsonHtlc{}
+		err := json.Unmarshal(htlc.CustomChannelData, jsonHtlc)
+		require.NoError(t, err)
+
+		for _, balance := range jsonHtlc.Balances {
+			totalAssetAmount += balance.Amount
+		}
+	}
+
+	// Due to rounding we allow up to 1 unit of error.
+	require.InDelta(t, assetAmount, totalAssetAmount, 1)
+}
+
+// assertPaymentHtlcAssets makes sure the payment with the given hash shows the
+// individual HTLCs that arrived for it and that they show the correct asset
+// amounts when decoded.
+func assertPaymentHtlcAssets(t *testing.T, node *HarnessNode, payHash []byte,
+	assetAmount uint64) {
+
+	ctxb := context.Background()
+	ctxt, cancel := context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+
+	stream, err := node.RouterClient.TrackPaymentV2(
+		ctxt, &routerrpc.TrackPaymentRequest{
+			PaymentHash:       payHash,
+			NoInflightUpdates: true,
+		},
+	)
+	require.NoError(t, err)
+
+	payment, err := stream.Recv()
+	require.NoError(t, err)
+	require.NotNil(t, payment)
+	require.NotEmpty(t, payment.Htlcs)
+
+	t.Logf("Asset payment: %v", toProtoJSON(t, payment))
+
+	var totalAssetAmount uint64
+	for _, htlc := range payment.Htlcs {
+		require.NotNil(t, htlc.Route)
+		require.NotEmpty(t, htlc.Route.CustomChannelData)
+
+		jsonHtlc := &rfqmsg.JsonHtlc{}
+		err := json.Unmarshal(htlc.Route.CustomChannelData, jsonHtlc)
+		require.NoError(t, err)
+
+		for _, balance := range jsonHtlc.Balances {
+			totalAssetAmount += balance.Amount
+		}
```

Comment:
This can potentially sum across multiple different assets (different asset IDs or groups).
Maybe we need to mod the doc to make that clear. Or rename `assetAmount` to `totalAssetAmounts` or something.
Same for the other assert.

Reply:
Good point, fixed.